.doom.d/config.org
* Table of Contents :toc:
- [[#theme][Theme]]
  - [[#font][Font]]
  - [[#style][Style]]
- [[#org-mode][Org-Mode]]
- [[#lsp-settings][LSP Settings]]
  - [[#java-support-lombok][Java Support (Lombok)]]
  - [[#gleam-support][Gleam Support]]
  - [[#angular-support][Angular Support]]
- [[#keymaps][Keymaps]]
  - [[#diagnostics][Diagnostics]]
- [[#fish-config][Fish Config]]
- [[#copilot][Copilot]]

* Theme
** Font
#+begin_src elisp
        (setq doom-font (font-spec :family "Monaspace Neon Var" :size 16))
#+end_src
** Style
#+begin_src elisp
        (setq doom-theme 'doom-ayu-dark)
        (setq display-line-numbers-type 'relative)
        (setq fancy-splash-image "~/.doom.d/dashboard.png")

#+end_src

#+RESULTS:
: ~/.doom.d/dashboard.png

* Org-Mode
#+begin_src elisp
(setq org-directory "~/org/")
#+end_src


* LSP Settings
** Java Support (Lombok)
#+begin_src elisp
(after! lsp-java

  (setq lombok-library-path (concat doom-data-dir "lombok.jar"))

  (unless (file-exists-p lombok-library-path)
    (url-copy-file "https://projectlombok.org/downloads/lombok.jar" lombok-library-path))

  (setq lsp-java-vmargs '("-XX:+UseParallelGC" "-XX:GCTimeRatio=4" "-XX:AdaptiveSizePolicyWeight=90" "-Dsun.zip.disableMemoryMapping=true" "-Xmx4G" "-Xms100m"))

  (push (concat "-javaagent:"
                (expand-file-name lombok-library-path))
        lsp-java-vmargs)
  )

#+end_src
** Gleam Support
#+begin_src elisp
(require 'lsp-mode)
(add-hook 'gleam-ts-mode-hook #'lsp)
(add-to-list 'auto-mode-alist '("\\.gleam\\'" . gleam-ts-mode))
#+end_src
** Angular Support
#+begin_src elisp
(setq lsp-clients-angular-language-server-command
      '("node"
        "/opt/homebrew/lib/node_modules/@angular/language-server"
        "--ngProbeLocations"
        "/opt/homebrew/lib/node_modules"
        "--tsProbeLocations"
        "/opt/homebrew/lib/node_modules"
        "--stdio"))
#+end_src

* Keymaps
** Diagnostics
#+begin_src elisp
(map! :n "]d" #'flycheck-next-error)
(map! :n "[d" #'flycheck-previous-error)
(map! :leader "cX" #'list-flycheck-errors)

(defun my/lsp-check-src-files ()
  "Open all files in the src directory and check for errors using LSP."
  (interactive)
  (let* ((project-root (projectile-project-root))
         (src-dir (expand-file-name "src" project-root))
         (src-files (directory-files-recursively src-dir "\\.\\(js\\|ts\\|html\\)$"))) ; Adjust the regex for your file types
    (dolist (file-path src-files)
      (with-current-buffer (find-file file-path)
        (lsp)))))

#+end_src

* Fish Config
#+begin_src elisp
(setq shell-file-name (executable-find "bash"))
(setq-default vterm-shell (executable-find "fish"))
(setq-default explicit-shell-file-name (executable-find "fish"))
#+end_src

* Copilot
#+begin_src elisp
(use-package! copilot
  :hook (prog-mode . copilot-mode)
  :bind (:map copilot-completion-map
              ("<tab>" . 'copilot-accept-completion)
              ("TAB" . 'copilot-accept-completion)
              ("C-TAB" . 'copilot-accept-completion-by-word)
              ("C-<tab>" . 'copilot-accept-completion-by-word)))
(use-package! gptel
  :config
  (setq! gptel-api-key "")
  (setq! gptel-model "gpt-4o"))
(setq! copilot-idle-delay 0.1)
#+end_src


.config/fish/config.org
#+title: Config
#+property: header-args :tangle config.fish

* Table of Contents :toc:
- [[#aliases][Aliases]]
- [[#variables][Variables]]
- [[#path][PATH]]
- [[#inits][inits]]

* Aliases
#+begin_src shell
alias ls="eza -al --git-repos"
alias cat='bat'
alias ec='emacsclient -c'
#+end_src

* Variables
"Editor": Standard editor to edit files
"fish_escape_delay_ms": no idea
"LLVM_PATH": Needed for homebrew
"NODE_EXTRA_CA_CERTS": company certs for nexus and archiva

#+begin_src shell
set -x EDITOR "emacsclient -c"
set -g fish_escape_delay_ms 10
set -x LLVM_PATH /opt/homebrew/opt/llvm/bin
set -x NODE_EXTRA_CA_CERTS "/Users/taradruffel/.workspace/repos/phoenix-core/optadata-gruppe-ODA-DC1-CA.crt"
set -x JAVA_HOME /opt/homebrew/opt/openjdk/
set -x USERNAME taradruffel
#+end_src

* PATH
#+begin_src shell
set -x PATH ~/.emacs.d/bin $PATH
set -x PATH /opt/homebrew/bin $PATH
set -x PATH /opt/homebrew/opt/grep/libexec/gnubin $PATH
set -x PATH /opt/homebrew/opt/openjdk/bin $PATH
#+end_src

* inits
#+begin_src shell
zoxide init fish | source
#+end_src


.doom.d/config.el
(setq doom-font (font-spec :family "Monaspace Neon Var" :size 16))

(setq doom-theme 'doom-ayu-dark)
(setq display-line-numbers-type 'relative)
(setq fancy-splash-image "~/.doom.d/dashboard.png")

(setq org-directory "~/org/")

(after! lsp-java

  (setq lombok-library-path (concat doom-data-dir "lombok.jar"))

  (unless (file-exists-p lombok-library-path)
    (url-copy-file "https://projectlombok.org/downloads/lombok.jar" lombok-library-path))

  (setq lsp-java-vmargs '("-XX:+UseParallelGC" "-XX:GCTimeRatio=4" "-XX:AdaptiveSizePolicyWeight=90" "-Dsun.zip.disableMemoryMapping=true" "-Xmx4G" "-Xms100m"))

  (push (concat "-javaagent:"
                (expand-file-name lombok-library-path))
        lsp-java-vmargs)
  )

(require 'lsp-mode)
(add-hook 'gleam-ts-mode-hook #'lsp)
(add-to-list 'auto-mode-alist '("\\.gleam\\'" . gleam-ts-mode))

(setq lsp-clients-angular-language-server-command
      '("node"
        "/opt/homebrew/lib/node_modules/@angular/language-server"
        "--ngProbeLocations"
        "/opt/homebrew/lib/node_modules"
        "--tsProbeLocations"
        "/opt/homebrew/lib/node_modules"
        "--stdio"))

(map! :n "]d" #'flycheck-next-error)
(map! :n "[d" #'flycheck-previous-error)
(map! :leader "cX" #'list-flycheck-errors)

(defun my/lsp-check-src-files ()
  "Open all files in the src directory and check for errors using LSP."
  (interactive)
  (let* ((project-root (projectile-project-root))
         (src-dir (expand-file-name "src" project-root))
         (src-files (directory-files-recursively src-dir "\\.\\(js\\|ts\\|html\\)$"))) ; Adjust the regex for your file types
    (dolist (file-path src-files)
      (with-current-buffer (find-file file-path)
        (lsp)))))

(setq shell-file-name (executable-find "bash"))
(setq-default vterm-shell (executable-find "fish"))
(setq-default explicit-shell-file-name (executable-find "fish"))

(use-package! copilot
  :hook (prog-mode . copilot-mode)
  :bind (:map copilot-completion-map
              ("<tab>" . 'copilot-accept-completion)
              ("TAB" . 'copilot-accept-completion)
              ("C-TAB" . 'copilot-accept-completion-by-word)
              ("C-<tab>" . 'copilot-accept-completion-by-word)))
(use-package! gptel
  :config
  (setq! gptel-api-key "")
  (setq! gptel-model "gpt-4o"))
(setq! copilot-idle-delay 0.1)
